core/src/main/java/com/linecorp/armeria/server/file/MediaTypeResolver.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.file;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A function used for determining the {@link MediaType} of a file based on its path.
+ */
+@UnstableApi
+public interface MediaTypeResolver {
+
+    /**
+     * Returns the default {@link MediaTypeResolver}.
+     */
+    static MediaTypeResolver ofDefault() {
+        return DefaultMediaTypeResolver.INSTANCE;
+    }
+
+    /**
+     * Resolves the {@link MediaType} of the file referred by the given {@code path}.
+     *
+     * @param path the path to the file to resolve its {@link MediaType}, e.g. {@code "/foo/bar.txt"}
+     *             or {@code "bar.txt"}.
+     * @return the resolved {@link MediaType}
+     */
+    @Nullable
+    MediaType guessFromPath(String path);
```

Comment:
Could we remove this API? I think we can think of `guessFromPath(String path)` as `guessFromPath(path, null)`?

Reply:
, and should we make this `@FuntionalInterface`?

Reply:
@klurpicolo We are about to release a new version. Addressed the comments myself. I really appreciate your enthusiastic contributions. Looking forward to your next!

Reply:
Many thanks!